server/src/graql/query/pattern/Pattern.java

```diff
@@ -65,19 +65,18 @@
     @CheckReturnValue
     Disjunction<Conjunction<Statement>> getDisjunctiveNormalForm();
 
+    @CheckReturnValue
+    Disjunction<Conjunction<Pattern>> getNegationDNF();
+
     /**
-     * @return this pattern negated
+     * @return true if this pattern corresponds to a negation pattern
      */
     @CheckReturnValue
-    Pattern negate();
+    default boolean isNegation(){ return false; }
 
     /**
-     * @return true if this pattern contains only positive patterns
+     * @return this Pattern as a Negation, if it is one.
      */
     @CheckReturnValue
-    default boolean isPositive(){
-        return getDisjunctiveNormalForm().getPatterns().stream()
-                .flatMap(p -> p.getPatterns().stream())
-                .allMatch(Pattern::isPositive);
-    }
+    default Negation<?> asNegation(){ throw new UnsupportedOperationException(); }
```

Comment:
We also don't need to "casting helper method". It hides the possibility of an exception being thrown. I would recommend doing an explicit act of casting when you need to. We've also started removing all methods like this in Grakn.